inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp

```diff
@@ -120,21 +120,35 @@ InferenceEngine::CNNNetwork dump_graph_as_ie_ngraph_net(const MKLDNNGraph &graph
     ngraph::NodeVector to_hold;
 
     auto get_inputs = [&] (const MKLDNNNodePtr & node) {
-        auto pr_edges = node->getParentEdges();
+        const auto& pr_edges = node->getParentEdges();
         ngraph::OutputVector inputs(pr_edges.size());
 
         for (int i = 0; i < pr_edges.size(); i++) {
-            auto edge = node->getParentEdgeAt(i);
-            int pr_port = edge->getInputNum();
-            int ch_port = edge->getOutputNum();
-            auto pr_node = edge->getParent();
+            const auto& edge = node->getParentEdgeAt(i);
+            const int pr_port = edge->getInputNum();
+            const int ch_port = edge->getOutputNum();
+            const auto& pr_node = edge->getParent();
 
             IE_ASSERT(node2layer.count(pr_node) == 1);
-            auto pr = node2layer[pr_node];
+            const auto& pr = node2layer[pr_node];
 
             inputs[ch_port] = pr->output(pr_port);
         }
 
+#ifdef CPU_DEBUG_CAPS
```

Comment:
After changing the contexts (now we want to not include constant nodes by default) I like more your first implementation then this :), since we cannot isolate DEBUG_CAPS piece of code anymore.
In you previous implementation you moved ch_port outside the loop, maybe we can just intputs.insert() instead of index access and do not preallocate memory for inputs?
With those comments I am just trying to make code less ugly, which is always the case when we use #ifdefs.

Reply:
Done